### PR TITLE
Remove magic string keys from TH builders.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Replaced the magic strings with extension methods that produce the same behavior as the string keys did.
- Added tests to validate new extension methods.

#1307

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Language/BoundAttributeDescriptorExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    public static class BoundAttributeDescriptorExtensions
+    {
+        public static string GetPropertyName(this BoundAttributeDescriptor descriptor)
+        {
+            descriptor.Metadata.TryGetValue(TagHelperBoundAttributeDescriptorBuilder.PropertyNameKey, out var propertyName);
+
+            return propertyName;
+        }
+
+        public static bool IsDefault(this BoundAttributeDescriptor descriptor)
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Language/BoundAttributeDescriptorExtensions.cs]: `IsDefaultKind`? I don't like this name

Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Language/BoundAttributeDescriptorExtensions.cs]: Ya I wasn't sold on `IsDefault` either but couldn't think of something better. Will rename.

File: src/Microsoft.AspNetCore.Razor.Language/TagHelperDescriptorBuilder.cs [hunk 2]
```diff
@@ -9,8 +9,8 @@ namespace Microsoft.AspNetCore.Razor.Language
 {
     public sealed class TagHelperDescriptorBuilder
     {
-        public static readonly string DescriptorKind = "ITagHelper";
-        public static readonly string TypeNameKey = "ITagHelper.TypeName";
+        internal const string DescriptorKind = "ITagHelper";
+        internal const string TypeNameKey = "ITagHelper.TypeName";
 
         private static ICollection<char> InvalidNonWhitespaceAllowedChildCharacters { get; } = new HashSet<char>(
             new[] { '@', '!', '<', '/', '?', '[', '>', ']', '=', '"', '\'', '*' });
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Language/TagHelperDescriptorBuilder.cs]: Not related to your changes. But is there somewhere we can put this collection and share it? Right now this is being duplicated in four different locations.

Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Language/TagHelperDescriptorBuilder.cs]: Will do as a follow up.

**Pull request merged**